wgpu-core/src/identity.rs

```diff
@@ -37,9 +37,12 @@ use std::{fmt::Debug, marker::PhantomData};
 #[derive(Debug, Default)]
 pub(super) struct IdentityValues {
     free: Vec<(Index, Epoch)>,
-    //sorted by Index
-    used: FastHashMap<Epoch, Vec<Index>>,
+    next_index: Index,
     count: usize,
+    // Sanity check: The allocation logic works under the assumption that we don't
+    // do a mix of allocating ids from here and providing ids manually for the same
+    // storage container.
+    allocate_ids: Option<bool>,
```

Comment:
Could we use an enum here? It would really help the readability of the uses to have an `enum IdSource { Allocated, UserAllocated }`